apps/docs/components/hooks/usePagination.md

```diff
@@ -0,0 +1,334 @@
+        >
+          <button
+            type="button"
+            :class="[
+              ' px-4 py-3 text-neutral-500 rounded-md hover:bg-primary-100 hover:text-primary-800 active:bg-primary-200 active:text-primary-900 ',
+              { '!text-neutral-900 hover:!text-primary-800 active:!text-primary-900': selectedPage === page },
+            ]"
+            :aria-label="`Page ${page} of ${totalPages}`"
+            :aria-current="selectedPage === page"
+            @click="onPageChange(page)"
+          >
+            {{ page }}
+          </button>
+        </div>
+      </li>
+      <li v-if="maxVisiblePages === 1 && selectedPage === 1">
+        <div class="flex pt-1 border-t-4 border-transparent">
+          <button
+            type="button"
+            class="px-4 py-3 rounded-md text-neutral-500 hover:bg-primary-100 hover:text-primary-800 active:bg-primary-200 active:text-primary-900"
+            :aria-label="`Page 2 of ${totalPages}`"
+            @click="onPageChange(2)"
+          >
+            2
+          </button>
+        </div>
+      </li>
+      <li v-if="endPage < totalPages - 1">
+        <div class="flex pt-1 border-t-4 border-transparent">
+          <button type="button" disabled aria-hidden="true" class="px-4 py-3 rounded-md text-neutral-500">...</button>
+        </div>
+      </li>
+      <li v-if="!pages.find((page) => page === totalPages)">
+        <div
+          :class="[
+            'flex pt-1 border-t-4 border-transparent ',
+            { 'font-medium border-t-4 !border-primary-500': selectedPage === totalPages },
+          ]"
+        >
+          <button
+            type="button"
+            class="px-4 py-3 rounded-md text-neutral-500 hover:bg-primary-100 hover:text-primary-800 active:bg-primary-200 active:text-primary-900"
+            :aria-current="totalPages === selectedPage"
+            @click="onPageChange(totalPages)"
+          >
+            {{ totalPages }}
+          </button>
+        </div>
+      </li>
+    </ul>
+    <SfButton
+      type="button"
+      aria-label="Go to next page"
+      :disabled="selectedPage >= totalPages"
+      variant="tertiary"
+      @click="next"
+    >
+      <template #suffix>
+        <SfIconChevronRight />
+      </template>
+    </SfButton>
+  </nav>
+</template>
+```
+<!-- end vue -->
+
+::::::
+
+::: slot api
+
+## Parameters
+
+| Name        | Type                  | Default value | Description                                  |
+| ---------   | --------------------- | ------------- | -------------------------------------------  |
+| totalItems  | number                |               | total number of items to be be displayed     |
+| currentPage | number                |  1            | current active page                          |
+| pageSize    | number                |  10           | number of items per page                     |
+| maxPages    | number                |  1            | maximum number of pages to display*          |
```

Comment:
```suggestion
| ---------   | --------------------- | ------------- | -------------------------------------------  |
| totalItems\*  | `number`                |               | total number of items to be be displayed     |
| currentPage | `number`                |  `1`            | current active page                          |
| pageSize    | `number`                |  `10`           | number of items per page                     |
| maxPages    | `number`                |  `1`            | maximum number of pages to display*          |
```
If im correct `totalItems` is required

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2535/commits/3ac7e43a8ab6b69cd07bd650bc5f0163b76f477b